HARK/distribution.py

```diff
@@ -1305,48 +1289,43 @@ def calc_expectation(dstn, func=lambda x: x, *args):
     Parameters
     ----------
     dstn : DiscreteDistribution
-        The N-valued distribution over which the function is to be evaluated.
+        The distribution over which the function is to be evaluated.
     func : function
         The function to be evaluated.
-        This function should take an array of size N x M.
-        It may also take other arguments *args
-        Please see numpy.apply_along_axis() for guidance on
-        design of func.
-        Defaults to identity function.
-    *args : scalar or np.array
-        One or more constants or arrays of input values for func,
-        representing the non-stochastic arguments.
-        The arrays must all have the same shape, and the expectation is computed
-        at f(dstn, args[0], args[1],...,args[M]).
+        This function should take an array of shape dstn.dim().
+        It may also take other arguments *args.
+    *args :
+        Other inputs for func, representing the non-stochastic arguments.
+        The the expectation is computed at f(dstn, *args).
 
     Returns
     -------
     f_exp : np.array or scalar
         The expectation of the function at the queried values.
         Scalar if only one value.
     """
-    N = dstn.dim()
-
-    dstn_array = np.column_stack(dstn.X)
-
-    if N > 1:
-        # numpy is weird about 1-D arrays.
-        dstn_array = dstn_array.T
-
-    f_query = np.apply_along_axis(func, 0, dstn_array, *args)
```

Comment:
@sbenthall keep the behavior of *args in mind when reviewing. It has changed!

Reply:
A quick search does suggest that `map` with lambda functions is quite slow.

Do you think there is a better, faster way to do it? The issue I was trying to get around is that `np.apply_along_axis` requires the function's inputs to be 1-d

Reply:
Seems like list comprehensions would work better:
https://stackoverflow.com/questions/1247486/list-comprehension-vs-map

Let me run some tests before merging.

Reply:
Looks like `np.apply_over_axes` is numpy's function to apply over more than one dimension

https://numpy.org/doc/stable/reference/generated/numpy.apply_over_axes.html#numpy.apply_over_axes

Would that fit?

Reply:
Thanks!

That certainly looks like something I could use. The only issue I see is that it does not accept extra arguments, like `np.apply_along_axis`. I could adapt it with a lambda function, not sure if it will make it lose too much performance.

I'll run some tests and report on Wed. Thanks!